parsl/dataflow/dflow.py

```diff
@@ -1271,6 +1271,11 @@ def cleanup(self) -> None:
         atexit.unregister(self.atexit_cleanup)
         logger.info("Unregistered atexit hook")
 
+        if DataFlowKernelLoader._dfk == self:
+            logger.info("Unregistering default DFK")
+            parsl.clear()
+            logger.info("Unregistered default DFK")
+
```

Comment:
If `==` check makes me wonder if there's a `!=` path to be considered?  Is it _possible_?  Would some kind of (at least a) log be warranted in the else-case?

Reply:
that path can exist, in the somewhat rare situation that you make a DFK by instantiating `DataFlowKernel` yourself, rather than using `parsl.load` -- in that case, the DataFlowKernelLoader (which manages this default DFK that comes from calling `parsl.load`) isn't aware of that DFK, and so it doesn't need to be told that it no longer needs to know about this DFK.

If you're doing this, you aren't going to get any of the defaulting behaviour (such as app invocations being sent automatically to that DFK) and you especially aren't going to hit the problem of not being able to get another DFK without clearing this one.

it's an artefact of history that users can make DFKs that way and I don't think I've really seen anyone do it since singleton-style loading was introduced in 2018 in da5935213490416d33af2ee77adb624db9f818e0

Reply:
(might be worth a debug message saying this isn't happening, but I don't think anything more)

Reply:
this test should be `is` not `==` though I think -- it's checking if the DFK objects are the same objects. (although right now, DFK `==` will behave the same as `is` because it is not overridden)